lib/cocoapods-core/podfile/target_definition.rb

```diff
@@ -753,17 +753,18 @@ def parse_configuration_whitelist(name, requirements)
       #         If :subspecs is the only key in the hash, the hash
       #         should be destroyed because it confuses Gem::Dependency.
       #
-      # @return [void]
+      # @return [TrueClass,FalseClass] Returns true if we added new subspecs
       #
       def parse_subspecs(name, requirements)
         options = requirements.last
-        return requirements unless options.is_a?(Hash)
+        return false unless options.is_a?(Hash)
 
         subspecs = options.delete(:subspecs)
         subspecs.each do |ss|
           store_pod("#{name}/#{ss}", *requirements.dup)
         end if subspecs
         requirements.pop if options.empty?
+        !subspecs.nil?
```

Comment:
That's not true by convention, if it documents it returns a Boolean, it just means you can only treat the return value as a Boolean. 

-Samuel E. Giddins

On Apr 6, 2015, at 8:03 AM, Kyle Fuller <notifications@github.com> wrote:

In lib/cocoapods-core/podfile/target_definition.rb:

>  
>          subspecs = options.delete(:subspecs)
>          subspecs.each do |ss|
>            store_pod("#{name}/#{ss}", *requirements.dup)
>          end if subspecs
>          requirements.pop if options.empty?
> +        !subspecs.nil?
Not exactly, if it's documented to return a Boolean it should 100% return a boolean.

> [].is_a?(TrueClass)
=> false
> [] == true
=> false
—
Reply to this email directly or view it on GitHub.